float32 exposure
---
bool success
